LogicLayer/Notifications/MailSender.cs

```diff
@@ -0,0 +1,115 @@
+            MailOptions = mailOptionsAccessor.Value;
+        }
+
+        public Task SendEmailConfirmationAsync(string email, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.EmailConfirmationHtml),
+                link,
+                "Email Confirmation");
+            return SendEmailAsync(email, notification);
+        }
+
+        public Task SendEventCancellationAsync(IEnumerable<string> emails, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.EventCancellationHtml),
+                link,
+                "Event Cancellation");
+            return SendEmailAsync(emails, notification);
+        }
+
+        public Task SendEventConfirmationAsync(IEnumerable<string> emails, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.EventConfirmationHtml),
+                link,
+                "Event Confirmation");
+            return SendEmailAsync(emails, notification);
+        }
+
+        public Task SendEventInvitationAsync(IEnumerable<string> emails, string link)
+        {
+            var notification = new Notification(GetHtmlPath(MailOptions.EventInvitationHtml), link, "Event Invitation");
+            return SendEmailAsync(emails, notification);
+        }
+
+        public Task SendFriendAcceptAsync(string email, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.FriendAcceptHtml),
+                link,
+                "Friend Invitation Accepted");
+            return SendEmailAsync(email, notification);
+        }
+
+        public Task SendFriendInvitationAsync(string email, string link)
+        {
+            var notification = new Notification(
+                GetHtmlPath(MailOptions.FriendInvitationHtml),
+                link,
+                "Friend Invitation");
+            return SendEmailAsync(email, notification);
+        }
+
+        public Task SendPasswordResetAsync(string email, string link)
+        {
+            var notification = new Notification(GetHtmlPath(MailOptions.PasswordResetHtml), link, "Password Reset");
+            return SendEmailAsync(email, notification);
+        }
+
+        private Task SendEmailAsync(IEnumerable<string> emails, Notification notification)
+            => Execute(MailOptions.SendGridApiKey, emails, notification);
+
+        private Task SendEmailAsync(string email, Notification notification)
+        {
+            var emails = new List<string>
+            {
+                email
+            };
+            return SendEmailAsync(emails, notification);
+        }
+
+        private Task Execute(string apiKey, IEnumerable<string> emails, Notification notification)
+        {
+            var client = new SendGridClient(apiKey);
+            var from = new EmailAddress("GameBoard.contact@gmail.com", "GameBoard");
```

Comment:
Two magic strings - we already have configuration for `MailNotifications`, move both there.

Reply:
Done